### PR TITLE
Java front-end: rule out multiple parallel instances of java_bytecode_languaget

### PR DESCRIPTION
**Pull request by smowton**
The front-end, unlike the C front-end, is not designed to be run on a partially populated symbol table.
Ensure we don't do that using an invariant in java_bytecode_languaget::typecheck, and forestall hitting
that invariant with a friendlier error message in the Java frontend programs.

Note jdiff did not require any change as it already insists on exactly two arguments.

@tautschnig I suggest this as an alternative to #4390

### CODE REVIEW

**Comment by tautschnig** [comment]
Would you mind also changing the `*symbol_table.lookup` in `java_entry_point.cpp` to something like `symbol_table.lookup_ref` or so such that at least a segmentation fault is avoided? I will then close #4390.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 32be85e18698fe76266277a3c8ecceff937775ad).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105821593

**Review by thk123 — APPROVED** [review]

**Comment by smowton** [comment]
@tautschnig https://github.com/diffblue/cbmc/pull/4437

**Comment by smowton** [comment]
@tautschnig do you approve now that #4437 is in?

**Review by tautschnig — APPROVED** [review]

**Comment by tautschnig** [comment]
> @tautschnig do you approve now that #4437 is in?

Sure, but my approval doesn't really count for Java stuff...

File: jbmc/src/jbmc/jbmc_parse_options.cpp [hunk 1]
```diff
@@ -530,6 +530,23 @@ int jbmc_parse_optionst::doit()
   stub_objects_are_not_null =
     options.get_bool_option("java-assume-inputs-non-null");
 
+  if(cmdline.args.empty())
+  {
+    error() << "Please provide a program to verify" << eom;
+    return 6;
```
Comment (peterschrammel) [inline jbmc/src/jbmc/jbmc_parse_options.cpp:536]: I'd use `CPROVER_EXIT_USAGE_ERROR` here

File: jbmc/src/jbmc/jbmc_parse_options.cpp [hunk 2]
```diff
@@ -530,6 +530,23 @@ int jbmc_parse_optionst::doit()
...
+
+  if(cmdline.args.size() != 1)
+  {
+    error() << "Only one .class, .jar or .gbf file should be directly "
+      "specified on the command-line. To force loading another another class "
+      "use '--java-load-class somepackage.SomeClass' or "
+      "'--lazy-methods-extra-entry-point somepackage.SomeClass.method' along "
+      "with '--classpath'"
+            << messaget::eom;
+    return 6;
```
Comment (peterschrammel) [inline jbmc/src/jbmc/jbmc_parse_options.cpp:547]: I'd use `CPROVER_EXIT_USAGE_ERROR` here

**Review by peterschrammel — APPROVED** [review]

**Comment by smowton** [comment]
Nothing else in `jbmc_parse_optionst` uses the named return values yet, so leaving as-is for now (change en masse welcome of course)

**Pull request merged**